pkg/resources/ratelimit/envoyconfig.go

```diff
@@ -50,3 +71,187 @@ func DeleteEnvoyConfigsInNamespace(ctx context.Context, client k8sclient.Client,
+	envoyconfig := &marin3rv1alpha1.EnvoyConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ec.name,
+			Namespace: ec.namespace,
+		},
+	}
+
+	envoyClusterResource := []marin3rv1alpha1.EnvoyResource{}
+	for _, cluster := range clusterResources {
+		jsonClusterResource, err := ResourcesToJSON(cluster)
+		if err != nil {
+			return fmt.Errorf("Failed to convert envoy rate limiting cluster configuration to JSON %v", err)
+		}
+
+		yamlClusterResource, err := yaml.JSONToYAML(jsonClusterResource)
+		if err != nil {
+			return fmt.Errorf("Failed to convert envoy rate limiting cluster JSON configuration to YAML %v", err)
+		}
+		envoyClusterResource = append(envoyClusterResource,
+			marin3rv1alpha1.EnvoyResource{
+				Name:  cluster.Name,
+				Value: string(yamlClusterResource),
+			},
+		)
+	}
+
+	envoyListenerResource := []marin3rv1alpha1.EnvoyResource{}
+	for _, listener := range listenerResources {
+		jsonListenerResource, err := ResourcesToJSON(listener)
+		if err != nil {
+			return fmt.Errorf("Failed to convert envoy rate limiting cluster configuration to JSON %v", err)
+		}
+
+		yamlListenerResource, err := yaml.JSONToYAML(jsonListenerResource)
+		if err != nil {
+			return fmt.Errorf("Failed to convert envoy rate limiting listener JSON configuration to YAML %v", err)
+		}
+		envoyListenerResource = append(envoyListenerResource,
+			marin3rv1alpha1.EnvoyResource{
+				Name:  listener.Name,
+				Value: string(yamlListenerResource),
+			},
+		)
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, client, envoyconfig, func() error {
+		owner.AddIntegreatlyOwnerAnnotations(envoyconfig, installation)
+		serialization := "yaml"
+		envoyconfig.Spec.NodeID = ec.nodeID
+		envoyconfig.Spec.Serialization = &serialization
+		envoyconfig.Spec.EnvoyResources = &marin3rv1alpha1.EnvoyResources{
+			Clusters:  envoyClusterResource,
+			Listeners: envoyListenerResource,
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Failed to create envoy config CR %v", err)
+	}
+	return nil
+}
+
+/**
+Creates envoy config cluster resource
+- name: clusterName
+	value: |
+		connectTimeout: 2s
+		loadAssignment:
+		clusterName: clusterName
+		endpoints:
+		- lbEndpoints:
+			- endpoint:
+				address:
+				socketAddress:
+					address: containerAddress
+					portValue: containerPort
+		name: apicast-ratelimit
+		type: STRICT_DNS
+**/
+func CreateClusterResource(containerAddress, clusterName string, containerPort int) *envoyapi.Cluster {
+
+	// Setting up cluster endpoints
+	clusterEndpoit := &envoycore.Address{
+		Address: &envoycore.Address_SocketAddress{
+			SocketAddress: &envoycore.SocketAddress{
+				Address:  containerAddress,
+				Protocol: envoycore.SocketAddress_TCP,
+				PortSpecifier: &envoycore.SocketAddress_PortValue{
+					PortValue: uint32(containerPort),
+				},
+			},
+		},
+	}
+
+	cluster := envoyapi.Cluster{
+		Name:                 clusterName,
+		ConnectTimeout:       ptypes.DurationProto(2 * time.Second),
```

Comment:
the only place time is used, but shouldn't it be a variable predefined along with others?

Reply:
The timeout value should be defined depending on the time the application receiving the request will take to respond, for now, we are defaulting it to 2 seconds but if the timeout should be changed in the future, we can pass it as one of the parameters of the `CreateClusterResource` function